src/Adapter/MSTest.TestAdapter/Execution/TypeCache.cs

```diff
@@ -694,12 +694,10 @@ private TestMethodInfo ResolveTestMethodInfo(TestMethod testMethod, TestClassInf
         return testMethodInfo;
     }
 
-    private TestMethodInfo ResolveTestMethodInfoForDiscovery(TestMethod testMethod, TestClassInfo testClassInfo)
+    private DiscoveryTestMethodInfo ResolveTestMethodInfoForDiscovery(TestMethod testMethod, TestClassInfo testClassInfo)
     {
         MethodInfo methodInfo = GetMethodInfoForTestMethod(testMethod, testClassInfo);
-
-        // Let's build a fake options type as it won't be used.
-        return new TestMethodInfo(methodInfo, testClassInfo, new(TimeoutInfo.FromTimeout(-1), null, false, null!));
```

Comment:
Building "fake" options type doesn't seem like a good idea. It's simply caller responsibility to be aware that these info are not correct and cannot be relied on. When reading the code, you have to go to references few levels up to understand and be sure that the "fake" options are indeed not used.

Instead, a new internal `DiscoveryTestMethodInfo` is introduced which holds only information relevant for discovery. It also doesn't calculate things that will end up not being used for discovery.